### ISSUE TITLE
文字识别 识别出了不存在的字典

### ISSUE THREAD

**Issue by a2174000000** [PaddlePaddle/PaddleOCR#4840]
yml配置如下:
Global:
  use_gpu: true
  epoch_num: 1000
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/rec/ic15/
  save_epoch_step: 3
  # evaluation is run every 2000 iterations after the 0th iteration
  eval_batch_step: [0, 2000]
  cal_metric_during_train: True
  pretrained_model:   #预训练模型 路径名 + 模型名 不带后缀
  checkpoints: C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/tools/output/rec/ic15/latest #继续训练模型
  save_inference_dir: ./
  use_visualdl: True
  infer_img:
  # for data or label process
  character_dict_path: C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/configs/Get_configs/ZiDian.txt #字典路径
  character_type: ch
  max_text_length: 1
  infer_mode: False
  use_space_char: False
  save_res_path: ./output/rec/predicts_ic15.txt
# C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/tools/output/rec/ic15/latest #预训练模型 路径名 + 模型名 不带后缀
Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.001
  regularizer:
    name: 'L2'
    factor: 0

Architecture:
  model_type: rec
  algorithm: CRNN
  Transform:
  Backbone:
    name: MobileNetV3
    scale: 0.5
    model_name: large
  Neck:
    name: SequenceEncoder
    encoder_type: rnn
    hidden_size: 96
  Head:
    name: CTCHead
    fc_decay: 0

Loss:
  name: CTCLoss

PostProcess:
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: SimpleDataSet
    data_dir: C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/configs/Get_configs/YangBen/
    label_file_list:
      - C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/configs/Get_configs/train_data/train.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 45, 45] #数组分别位 【通道数,高度,宽度】
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: True #每个epoch是否将数据集顺序打乱
    batch_size_per_card: 256 #训练时单卡batch size
    drop_last: True #是否丢弃因数据集样本数不能被 batch_size 整除而产生的最后一个不完整的mini-batch
    num_workers: 8 #用于加载数据的子进程个数，若为0即为不开启子进程，在主进程中进行数据加载
    use_shared_memory: False

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/configs/Get_configs/YangBen/
    label_file_list:
      - C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/configs/Get_configs/train_data/test.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 45, 45] #数组分别位 【通道数,高度,宽度】
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: False #每个epoch是否将数据集顺序打乱
    drop_last: False #训练时单卡batch size
    batch_size_per_card: 256 #是否丢弃因数据集样本数不能被 batch_size 整除而产生的最后一个不完整的mini-batch
    num_workers: 8 #用于加载数据的子进程个数，若为0即为不开启子进程，在主进程中进行数据加载
    use_shared_memory: False

采用的ic15配置稍作修改 我字典文本 全是一位数的汉字 但是具体测试的时候返回的结果是(因为我把训练好的已经删除了 所以发的是重新训练识别的)

[2021/12/06 00:02:40] root INFO: Architecture :
[2021/12/06 00:02:40] root INFO:     Backbone :
[2021/12/06 00:02:40] root INFO:         model_name : large
[2021/12/06 00:02:40] root INFO:         name : MobileNetV3
[2021/12/06 00:02:40] root INFO:         scale : 0.5
[2021/12/06 00:02:40] root INFO:     Head :
[2021/12/06 00:02:40] root INFO:         fc_decay : 0
[2021/12/06 00:02:40] root INFO:         name : CTCHead
[2021/12/06 00:02:40] root INFO:     Neck :
[2021/12/06 00:02:40] root INFO:         encoder_type : rnn
[2021/12/06 00:02:40] root INFO:         hidden_size : 96
[2021/12/06 00:02:40] root INFO:         name : SequenceEncoder
[2021/12/06 00:02:40] root INFO:     Transform : None
[2021/12/06 00:02:40] root INFO:     algorithm : CRNN
[2021/12/06 00:02:40] root INFO:     model_type : rec
[2021/12/06 00:02:40] root INFO: Eval :
[2021/12/06 00:02:40] root INFO:     dataset :
[2021/12/06 00:02:40] root INFO:         data_dir : C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/configs/Get_configs/YangBen/
[2021/12/06 00:02:40] root INFO:         label_file_list : ['C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/configs/Get_configs/train_data/test.txt']
[2021/12/06 00:02:40] root INFO:         name : SimpleDataSet
[2021/12/06 00:02:40] root INFO:         transforms :
[2021/12/06 00:02:40] root INFO:             DecodeImage :
[2021/12/06 00:02:40] root INFO:                 channel_first : False
[2021/12/06 00:02:40] root INFO:                 img_mode : BGR
[2021/12/06 00:02:40] root INFO:             CTCLabelEncode : None
[2021/12/06 00:02:40] root INFO:             RecResizeImg :
[2021/12/06 00:02:40] root INFO:                 image_shape : [3, 45, 45]
[2021/12/06 00:02:40] root INFO:             KeepKeys :
[2021/12/06 00:02:40] root INFO:                 keep_keys : ['image', 'label', 'length']
[2021/12/06 00:02:40] root INFO:     loader :
[2021/12/06 00:02:40] root INFO:         batch_size_per_card : 256
[2021/12/06 00:02:40] root INFO:         drop_last : False
[2021/12/06 00:02:40] root INFO:         num_workers : 8
[2021/12/06 00:02:40] root INFO:         shuffle : False
[2021/12/06 00:02:40] root INFO:         use_shared_memory : False
[2021/12/06 00:02:40] root INFO: Global :
[2021/12/06 00:02:40] root INFO:     cal_metric_during_train : True
[2021/12/06 00:02:40] root INFO:     character_dict_path : C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/configs/Get_configs/ZiDian.txt
[2021/12/06 00:02:40] root INFO:     character_type : ch
[2021/12/06 00:02:40] root INFO:     checkpoints : C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/tools/output/rec/ic15/latest
[2021/12/06 00:02:40] root INFO:     debug : False
[2021/12/06 00:02:40] root INFO:     distributed : False
[2021/12/06 00:02:40] root INFO:     epoch_num : 1000
[2021/12/06 00:02:40] root INFO:     eval_batch_step : [0, 2000]
[2021/12/06 00:02:40] root INFO:     infer_img : C:/Users/Administrator/Desktop/cs/
[2021/12/06 00:02:40] root INFO:     infer_mode : False
[2021/12/06 00:02:40] root INFO:     load_static_weights : False
[2021/12/06 00:02:40] root INFO:     log_smooth_window : 20
[2021/12/06 00:02:40] root INFO:     max_text_length : 1
[2021/12/06 00:02:40] root INFO:     pretrained_model : ./tools/output/rec/ic15/best_accuracy
[2021/12/06 00:02:40] root INFO:     print_batch_step : 10
[2021/12/06 00:02:40] root INFO:     save_epoch_step : 3
[2021/12/06 00:02:40] root INFO:     save_inference_dir : ./
[2021/12/06 00:02:40] root INFO:     save_model_dir : ./output/rec/ic15/
[2021/12/06 00:02:40] root INFO:     save_res_path : ./output/rec/predicts_ic15.txt
[2021/12/06 00:02:40] root INFO:     use_gpu : True
[2021/12/06 00:02:40] root INFO:     use_space_char : False
[2021/12/06 00:02:40] root INFO:     use_visualdl : True
[2021/12/06 00:02:40] root INFO: Loss :
[2021/12/06 00:02:40] root INFO:     name : CTCLoss
[2021/12/06 00:02:40] root INFO: Metric :
[2021/12/06 00:02:40] root INFO:     main_indicator : acc
[2021/12/06 00:02:40] root INFO:     name : RecMetric
[2021/12/06 00:02:40] root INFO: Optimizer :
[2021/12/06 00:02:40] root INFO:     beta1 : 0.9
[2021/12/06 00:02:40] root INFO:     beta2 : 0.999
[2021/12/06 00:02:40] root INFO:     lr :
[2021/12/06 00:02:40] root INFO:         learning_rate : 0.001
[2021/12/06 00:02:40] root INFO:     name : Adam
[2021/12/06 00:02:40] root INFO:     regularizer :
[2021/12/06 00:02:40] root INFO:         factor : 0
[2021/12/06 00:02:40] root INFO:         name : L2
[2021/12/06 00:02:40] root INFO: PostProcess :
[2021/12/06 00:02:40] root INFO:     name : CTCLabelDecode
[2021/12/06 00:02:40] root INFO: Train :
[2021/12/06 00:02:40] root INFO:     dataset :
[2021/12/06 00:02:40] root INFO:         data_dir : C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/configs/Get_configs/YangBen/
[2021/12/06 00:02:40] root INFO:         label_file_list : ['C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/configs/Get_configs/train_data/train.txt']
[2021/12/06 00:02:40] root INFO:         name : SimpleDataSet
[2021/12/06 00:02:40] root INFO:         transforms :
[2021/12/06 00:02:40] root INFO:             DecodeImage :
[2021/12/06 00:02:40] root INFO:                 channel_first : False
[2021/12/06 00:02:40] root INFO:                 img_mode : BGR
[2021/12/06 00:02:40] root INFO:             CTCLabelEncode : None
[2021/12/06 00:02:40] root INFO:             RecResizeImg :
[2021/12/06 00:02:40] root INFO:                 image_shape : [3, 45, 45]
[2021/12/06 00:02:40] root INFO:             KeepKeys :
[2021/12/06 00:02:40] root INFO:                 keep_keys : ['image', 'label', 'length']
[2021/12/06 00:02:40] root INFO:     loader :
[2021/12/06 00:02:40] root INFO:         batch_size_per_card : 256
[2021/12/06 00:02:40] root INFO:         drop_last : True
[2021/12/06 00:02:40] root INFO:         num_workers : 8
[2021/12/06 00:02:40] root INFO:         shuffle : True
[2021/12/06 00:02:40] root INFO:         use_shared_memory : False
[2021/12/06 00:02:40] root INFO: train with paddle 2.2.0 and device CUDAPlace(0)
W1206 00:02:40.405678 10244 device_context.cc:447] Please NOTE: device: 0, GPU Compute Capability: 7.5, Driver API Version: 11.5, Runtime API Version: 10.2
W1206 00:02:40.409682 10244 device_context.cc:465] device: 0, cuDNN Version: 7.6.
[2021/12/06 00:02:45] root INFO: resume from C:/Users/Administrator/Desktop/PaddleOCR-release-2.3/tools/output/rec/ic15/latest
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/26192ee326c735f1efcf1bca3b375bfc.jpg
[2021/12/06 00:02:45] root INFO:         result: 世卢   0.006314099
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/_0243e570702b2763712fe552117d4a3a.jpg
[2021/12/06 00:02:45] root INFO:         result: 峰     0.006577249
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/啊_0ac6e515e2ea77010eb3d912b96f5742.jpg
[2021/12/06 00:02:45] root INFO:         result: 唇     0.009107739
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/啊_0ce688d1c40c9569aad40726ad59cf31.jpg
[2021/12/06 00:02:45] root INFO:         result:        nan
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/啊_0e8360a1ab2844616ef3aade2a4e96f4.jpg
[2021/12/06 00:02:45] root INFO:         result: 峰     0.007773535
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/啊_1b3f3cb3b3c1c36d2bb15ff0875331d4.jpg
[2021/12/06 00:02:45] root INFO:         result: 峰     0.007888668
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/啊_1b87c28f7b9df2189da606afa530b9af.jpg
[2021/12/06 00:02:45] root INFO:         result: 峰     0.007757437
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/啊_1b9b6cfb42f3de4ace0324edada013c2.jpg
[2021/12/06 00:02:45] root INFO:         result: 唇     0.008316142
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/啊_1e926691bd2974772cac91ef252fa0f5.jpg
[2021/12/06 00:02:45] root INFO:         result: 峰     0.007019079
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/啊_2b7951cb5021d1b06ec26362b77b8388.jpg
[2021/12/06 00:02:45] root INFO:         result: 唇     0.009594217
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/阿_012256b9b8af89a104d1bdd91f750be1.jpg
[2021/12/06 00:02:45] root INFO:         result: 掩     0.00504731
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/阿_08432de0cba2f36db4133cb25b74e650.jpg
[2021/12/06 00:02:45] root INFO:         result: 润趋   0.0036844662
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/阿_0883dd3cc1061e5067ffc6a86f26c3ad.jpg
[2021/12/06 00:02:45] root INFO:         result: 渴     0.0048181443
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/阿_1400893d510ee229f8fd8a516466fd23.jpg
[2021/12/06 00:02:45] root INFO:         result: 弯     0.0069923294
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/阿_14111b462924f3937a3e364ec0c078f6.jpg
[2021/12/06 00:02:45] root INFO:         result: 趋局   0.004366738
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/阿_1516faa58ea006f331f967a988ed714c.jpg
[2021/12/06 00:02:45] root INFO:         result: 弯     0.0071730535
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/阿_20734c611071edee207f751685740245.jpg
[2021/12/06 00:02:45] root INFO:         result: 版典   0.0034746758
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/阿_25821a83281e06b532d0f98bd0b2205c.jpg
[2021/12/06 00:02:45] root INFO:         result: 美     0.004182768
[2021/12/06 00:02:45] root INFO: infer_img: C:/Users/Administrator/Desktop/cs/阿_266472088be452ae382ac22491705985.jpg
[2021/12/06 00:02:45] root INFO:         result: 弯     0.006875559
[2021/12/06 00:02:45] root INFO: success!

这让我特别的纳闷 我字典中没有 两位长度的汉字 还有就是 PaddleOCR 分割字典的时候会在首位 加入一个 blank 的数组么

**Comment by zhangyingying520** [comment]
可能没有正确加载你更换后的字典

**Comment by MissPenguin** [comment]
字典本来就都是单个字的，和识别结果是单个还是多个字没关系。我看你的配置里改了个参数：
![image](https://user-images.githubusercontent.com/25809855/145211042-01342222-8a0a-4762-9c2b-44fbe7cbafde.png) 你的训练数据都是单字吗，这会导致只有1个字的数据才会进入训练，超过1个字就过掉了，可能你的训练数据根本都没训。最后你贴出来的预测结果看着置信度都很低，可能都是随机结果。

**Comment by a2174000000** [comment]
> 

感谢你的回答 问题已经已经解决了 问题出在
parser.add_argument("--rec_algorithm", type=str, default='CRNN')
    parser.add_argument("--rec_model_dir", type=str)
    parser.add_argument("--rec_image_shape", type=str, default="3, 32, 50")
    parser.add_argument("--rec_char_type", type=str, default='ch')
    parser.add_argument("--rec_batch_num", type=int, default=1)
    parser.add_argument("--max_text_length", type=int, default=5)

但是他为什么能识别出多个 我也不不知道是什么原因 至于预测结果看是置信度很低 我在上文中已经注释了 我把之前训练置信度高的删了 改了一点配置又去训练的日志 但还是出现多字
